libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -975,6 +975,11 @@ - (SFAuthErrorHandlerList *)populateDefaultAuthErrorHandlerList
                                                    [weakSelf execFailureBlocks];
                                                    return YES;
                                                }
+                                               if (error.code == kSFOAuthErrorJWTInvalidGrant) {
+                                                   [weakSelf log:SFLogLevelWarning format:@"JWT swap failed due to invalid grant.  Error code: %ld", (long)error.code];
+                                                   [weakSelf execFailureBlocks];
```

Comment:
@kchitalia  So authenticationManager is the delegate for oauth failure.  And here as you can see, it will call "execFailureBlocks" which will eventually call [SalesforceSDKManager sharedManager].launchErrorAction,  so that user can just implement their desired flow in launchErrorAction..